# BINF6310:
    To better utilize and develop intentional and deliberate scaffolding, scope, and sequence
    within the bioinformatics program, the proposal for BINF6310 is both a curricular redesign and
    reconsolidation of core bioinformatics courses--namely BINF6308 and BINF6309. Historically, there
    has been significant overlap between the BINF6200 (Programming in Bioinformatics), BINF6308, BINF6309. 
    This, coupled with leniant pre-reqs, led to a pedagogical and androgogical morass that often plagued 
    start of term and would leach from didactic momentum. Furthermore, the structure of a two-sememster, introductor
    survey course of bioinformatics is not necessarily reflected in other comparable bioinformatics. That is,
    the same content of BINF6308 and BINF6309 are often covered within a single 4-credit course at comparable
    institutions (e.g., University of Michigan's BIOINF-527: Introduction to Bioinformatics & Computational Biology, 
    Georgia Tech's BIOL 6150: Genomics and Applied Bioinformatics, and University of Wisconsin-Madison's BMI/COMP SCI
    576: Introduction to Bioinformatics). Therefore, we propose that the new course, BINF6310, will reconsolidate and 
    restructure the content of BINF630 & BINF6309 as follows:

The final project of the course is to be a small group project wherein the students will be tasked to:
    1. Identify a peer-reviewed journal article on a bioinformatic analysis technique, software, or line of inquiry
    2. Critically read the paper and perform a mid-term "journal club"-style presentation of the paper
    3. Find or select an appropriate dataset from a publically available dataset that can either be used to:
        a. corroborate the findings of the paper or
        b. replicate the finding of the paper
    4. Replicate the pipeline from their selected paper--within reason and as relevant to the course topic
    5. Analyze the results
    6. Individually write up the process in the form of a publication
    7. As a group, present their project to the class

## week 01: Molecular biology and genetics 1
## week 02: Molecular biology and genetics 2
    * These two weeks will be a deep dive into the fundamental molecular biology and genetics 
    of bioinformatics. Unique genetic concepts relevant to bioinformatics (e.g., genetic differences
    between eukaryotes and prokaryotes, chromatic accessibility, orgranization & structure of the genome, etc.) 
    will be covered in detail in addition to experimental design and model organism selection.
    
    * This section serves two additional purposes: 1) to develop additional soft-skills in our students regarding
    critical reading and evaluation of peer-reviewed literature through "journal-club" assignments of seminal
    papers in the field and 2) allow students taking BINF6200 as a co-requisite an opportunity to develop
    digital and computational literacy before introducing extensions of that scaffolding in this course.
## week 03: qPCR, dd-qPCR*, molecular techniques
    * This module will cover the above topics, but specifically reflecting on the bioinformatic concepts and biases
    that are bundled with these techniques. An example of these concepts are FISH/GFP flourescence, ligation, ligation and elongation,
    DNA repair, and primer/probe design.
    
    * This section couples with the previous two regarding a strong biological and genetic focus, specifically regarding bench work.
    However, the concepts of this section develop the adragogical scaffolding necessary to introduce topics like probe design for 
    targetted genomic assays and elongation and flourescence--which are critical in fundamentally understanding various next-generation
    sequencing technologies.
    
    \* indicates discussions on single cell applications of the technology
## week 04: Sequencing 1: Sanger, microarrays, and NGS 1*
    * The section covers the history of the Human Genome Project and early sequencing technologies. However, with the conceptual background
    in place from the previous sections, students will be introduced to methods and techniques used in library preparation of biological samples,
    massively-parallel sequencing, and limitations of the technologies through a bilogical lens.
    
    * This section will be the first introduction to students to Linux, BASH, and the command line interface. The students will be take part
    in active learning exercises related to the transversal of folder structures, manipulation of files, and searching & analysis of large 
    text-based files.
    
    \* indicates discussions on single cell applications of the technology
## week 05: Sequencing 1: NGS 2* and TGS (FASTQ)
    * This section will further expand on next-generation sequencing and its various forms as well as fully introduce students
    to cutting edge third generation sequencing technologies like Oxford Nanopore and PacBio SMRT sequencing. To continue the theme of
    cutting edge research, the applications of these technologies to single-cell sequencing will be explored further.
    
    * This section will formally introduce students to their first standard format file type: FASTQ files. Using the scaffolding in place
    from BINF6200, students will use simple conditionals and iterative approaches to programmatically analyze FASTQ files
    
    \* indicates discussions on single cell applications of the technology
## week 06: References, contigs, and alignments (SAM/BAM/CRAM)
    * This section serves to establish the understanding of reference assemblies and the various approaches to aligning sequencing data to them.
    Additionally, students will actively explore the concepts of multiple sequence alignment, local vs global alingment, and the effect the biological
    sequence has on application of these methods. That is, how does RNA alignment differ from DNA. Students will be tasked with developing
    rudimenary pipelines to perform straightforward quality control on sequenced data, alignment of the data to a standard reference, and analysis of
    the results.
    
    * This section will also introduce students to three core standard file formats: SAM/BAM/CRAM. Students will use contemporary tools to programmatically 
    parse the esulting files generated from their alignment pipeline.
    
    *Students will be introduced to simple bash scripting
## week 07: Public datasets: NCBI, GEO, SRA, Ensembl, ENCODE, GTeX, and TCGA
    * This section will be a very broad overview of extant resources. These resources will be used to help annotate data generated in future sections of
    the course. As such, the students will be introduced to ways to visualize data through the use of genome browsers.
BREAK
## week 08: HPC/remote, containers, and Pipelines (Nextflow)
    * This section serves to introduce students to advanced concepts in remote/cloud high-performance computing, container technologies (e.g., docker & Singularity), and a brief introduction to pipeline development and maintenance via Nextflow, CWL, or snakemake.
    
    * The students will perform active learning tasks of programmatic data retrieval from public databases (from the previous section) and expand their
    initial computational tasks of quality control and alignment scripts to more deliberate and sustainable pipeline development.
    
    * An important soft-skill that will be developed in this section is that of HPC/Remote computing ettiquette. That is, how to use these resources
    responsibly and courteously within a research community.
## week 09: WGS, WES, ChIP-seq, and shotgun*
    * This section will provide an introduction to whole genome sequencing (WGS), Whole Exome Sequencing (WES), ChIP-seq (and similar techniques), and 
    the concept of shotgun sequencing in general. Aside from the survey of the technologies and techniques listed, this section recontextualizes the
    analysis back to the biological question. That is, "what is being sequenced and what can be answered with it?" Through this reflection, students will
    explore some of the varying techniques 
    
    * Students will be tasked with aligning WGS data to a standard reference and explore the topics of unaligned sequences, aneuploidy, and diseases assoicated
    with genome variation
    \* indicates discussions on single cell applications of the technology
## week 10: RNA-seq: 16s and the transcriptome*
    *Similar to the previous section, this section will introduce the students to RNA sequencing and its many variants. The concept of the transcriptome--and
    how it differs from the genome and exome will be explored in-depth.
    
    * Students will align relevant data and explore the topics of splice isoforms, lincRNA, lncRNA, 3' bias, and 16s sequencing
    
    \* indicates discussions on single cell applications of the technology
## week 11: MSA, BLAST, ortho-/paralogs
    * This section will cover the basics of BLAST and its different types, how it relates to functional homology, and why orthologs matter.
    Furthermore, this section will explore the limitations of multiple sequence alignment and its applications to phylogenomics, phylogenetic trees,
    and comparative genomics and transcriptomics.
## week 12: Survey of Variants and Cancer Genomics (VCF, GTF, GFF3)
    * This section will cover the complex nature of germline and somatic mutations and survey the landscape of different types of variants such as 
    Single nucleotide polymorphisms/variants, copy number variants, translocations, transversions, duplications, indels, fusion events, and the 
    nomenclature of variants relative to the clinical setting.
    
    * The last set of standard file formats will be introduced. These will include variant call format (VCF), general feature format version 3 (GFF3), and 
    gene transfer format (GTF). Students will be tasked with programmatic parsing of these files using relevant tools, basic statisical analysis of the results,
    and visualization of variation on genome browsers.
## week 13: (SPECIAL TOPIC) Systems Biology and agent-based modeling
    * This section will be a special topics section that will partner with our research faculty to explore a given concept. The first
    section of this course will explore the concept of Systems Biology and will be guided through a guest lecture by our research/faculty, academic partner,
    or industry partner.
## week 14: Work week (Final project on group presenations of specific software/pipeline)
